commit df1b3e87793175558f0647ca15d894f18fd81158 (refs/changes/23/2923/3)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-11-07T23:12:02+00:00 (1 year, 11 months ago)
    
    TOOLS-1923 VOLAPI node-sdc-clients tests should skip instead of exiting when VOLAPI not present
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/test/runtests b/test/runtests
index 01453f5..9c43144 100755
--- a/test/runtests
+++ b/test/runtests
@@ -116,6 +116,7 @@ if [[ -n "$opt_headnode" ]]; then
     export AMON_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json amon_admin_ips | cut -d, -f1')
     export CNAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json cnapi_admin_ips | cut -d, -f1')
     export CA_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json ca_admin_ips | cut -d, -f1')
+    export IMGAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json imgapi_admin_ips | cut -d, -f1')
     export NAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json napi_admin_ips | cut -d, -f1')
     export VMAPI_IP=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json vmapi_admin_ips | cut -d, -f1')
     export UFDS_ADMIN_UUID=$(ssh $opt_headnode 'bash /lib/sdc/config.sh -json | json ufds_admin_uuid')
diff --git a/test/volapi.test.js b/test/volapi.test.js
index 61a6e13..47b6966 100644
--- a/test/volapi.test.js
+++ b/test/volapi.test.js
@@ -17,11 +17,6 @@ var test = require('tape');
 assert.ok(typeof (process.env.NAPI_IP) === 'string' &&
     process.env.NAPI_IP !== '', 'NAPI_IP env var must be a non-empty string');
 
-if (!process.env.VOLAPI_IP) {
-    console.log('Could not find VOLAPI core service IP address, skipping.');
-    process.exit(0);
-}
-
 var NAPI = require('../lib/index').NAPI;
 
 var ADMIN_UUID = process.env.UFDS_ADMIN_UUID;
@@ -36,9 +31,16 @@ var VOLAPI = require('../lib/index').VOLAPI;
 var VOLAPI_IP = process.env.VOLAPI_IP;
 var VOLAPI_URL = 'http://' + VOLAPI_IP;
 
+var testOps = {
+    skip: !process.env.VOLAPI_IP
+};
 var volApiClient;
 
-test('volapi', function (tt) {
+if (testOps.skip) {
+    console.error('# Could not find VOLAPI core service IP address, skipping.');
+}
+
+test('volapi', testOps, function (tt) {
     var volumeResOwnerUuid;
     var volumeResUuid;
 
